lib/stats/stats-control.c

```diff
@@ -53,10 +47,16 @@ _reset_counter_if_needed(StatsCluster *sc, gint type, StatsCounterItem *counter,
     case SC_TYPE_QUEUED:
       return;
     default:
-      _reset_counter(sc, type, counter, user_data);
+      stats_counter_set(counter, 0);
     }
 }
 
+void
+stats_control_reset_counter_if_needed(StatsCluster *sc, StatsCounterItem *counter)
```

Comment:
This and the actual implementation `_reset_counter_if_needed()` should go to the `StatsCluster` unit:
`stats_cluster_reset_counter_if_needed(self, c)`

Reply:
i couldn't really decide where it really belongs, so thanks for helping to dispel my doubts